app/assets/javascripts/basetypes/Interval.js

```diff
@@ -7,10 +7,28 @@ function Interval(key, options) {
 Interval.prototype = Object.create(mongoose.SchemaType.prototype);
 
 Interval.prototype.cast = (interval) => {
-  if (typeof interval.low === 'undefined') {
+  if (typeof interval.low === 'undefined' || interval.low === null) {
     throw new Error(`Interval: ${interval} does not have a low value`);
   }
-  return new cql.Interval(interval.low, interval.high, interval.lowClosed, interval.highClosed);
+  const casted = new cql.Interval(interval.low, interval.high, interval.lowClosed, interval.highClosed);
+
+  // Cast Low and High values to Quantities if it is a quantity
+  if (casted.low && casted.low.unit && casted.low.value) {
```

Comment:
Is this assuming that only Quantities will have a `unit` and `value` and that is how this is determining that `casted` is a Quantity? Could you check the type in some way instead?

Reply:
I don't think so. The JSON objects being casted are just that--JSON with no inherent type except being an object. Having a unit and value is the only way we can really identify them as Quantities.